site/static/compare/script.js

```diff
@@ -434,39 +437,107 @@ const SummaryCount = {
+                <td><SummaryPercentValue :value="summary.regressions.average" /></td>
+                <td><SummaryCount :cases="summary.regressions.count" :benchmarks="summary.regressions.benchmarks" /></td>
+            </template>
+            <template v-else>
+              <td colspan="3" style="text-align: center;">No regressions</td>
+            </template>
         </tr>
         <tr class="negative">
-            <td title="Improvements">✅</td>
-            <td><SummaryRange :range="cases.improvements.range" /></td>
-            <td><SummaryPercentValue :value="cases.improvements.average" /></td>
-            <td><SummaryCount :cases="cases.improvements.count" :benchmarks="cases.improvements.benchmarks" /></td>
+            <td title="Improvements" v-if="withLegend">✅</td>
+            <template v-if="summary.improvements.count !== 0">
+                <td><SummaryRange :range="summary.improvements.range" /></td>
+                <td><SummaryPercentValue :value="summary.improvements.average" /></td>
+                <td><SummaryCount :cases="summary.improvements.count" :benchmarks="summary.improvements.benchmarks" /></td>
+            </template>
+            <template v-else>
+              <td colspan="3" style="text-align: center;">No improvements</td>
+            </template>
         </tr>
         <tr>
-            <td title="All changes">❌,✅</td>
-            <td><SummaryRange :range="cases.all.range" /></td>
-            <td><SummaryPercentValue :value="cases.all.average" /></td>
-            <td><SummaryCount :cases="cases.all.count" :benchmarks="cases.all.benchmarks" /></td>
+            <td title="All changes" v-if="withLegend">❌,✅</td>
+            <td><SummaryRange :range="summary.all.range" /></td>
+            <td><SummaryPercentValue :value="summary.all.average" /></td>
+            <td><SummaryCount :cases="summary.all.count" :benchmarks="summary.all.benchmarks" /></td>
         </tr>
     </tbody>
 </table>
 `,
     components: {SummaryRange, SummaryPercentValue, SummaryCount}
 });
+
+app.component("aggregations", {
+    props: {
+        cases: Array
+    },
+    template: `
+<div>
+  <div class="aggregation-section">
+    <div class="header">Profile</div>
+    <div class="groups">
+      <div class="group" v-for="profile in ['check', 'debug', 'opt', 'doc']">
+        <div class="group-header">{{ profile }}</div>
+        <summary-table :summary="calculateSummary('profile', profile)" :withLegend="false"></summary-table>
+      </div>
+    </div>
+  </div>
+  <div class="aggregation-section">
+    <div class="header">Scenario</div>
+    <div class="groups">
+      <div class="group" v-for="scenario in ['full', 'incr-full', 'incr-unchanged', 'incr-patched']">
+        <div class="group-header">{{ scenario }}</div>
+        <summary-table :summary="calculateSummary('scenario', scenario)" :withLegend="false"></summary-table>
+      </div>
+    </div>
+  </div>
+  <div class="aggregation-section">
```

Comment:
Probably not worth it but we can reduce the amount of code here by making each aggregation section its own component.